### DISCUSSION TITLE
expand_op in wiki seems typo

### DISCUSSION THREAD

**Discussion by erickaoshoulin** [kedartatwawadi/stanford_compression_library#50] [Q&A] [answered]
Appreciate the ANS tutorial! Thank you guys.

In wiki:
2: decode-step-constraint: During the expand_state we repeatedly perform the state expansion operation:

start -> x_0

> expand_op -> x_i = x_{i-1} + bit

It should be 

> expand_op -> x_i = **2***x_{i-1} + bit

right?

**Comment by kedartatwawadi** [comment, marked as answer]
Yes that is right! Thanks for informing about the typo. 
I fixed this in the wiki. Let us know in case you find other typos :). 

Also, the rANS and tANS implementations are now merged to the main branch:
1. https://github.com/kedartatwawadi/stanford_compression_library/blob/main/compressors/rANS.py
2. https://github.com/kedartatwawadi/stanford_compression_library/blob/main/compressors/tANS.py

**Reply by erickaoshoulin** [reply]
Hello Kedar,
I think it's another typo In wiki tANS encode part:

Now, lets say, freq[s] <= 2^y < (2*freq[s] - 1). Then we know the following:

> For any x_shrunk in [2^y, 2*2^y - 1], we need to output r+1-**e** bits.
> For any x_shrunk in [2^{y-1}, 2*2^{y-1} - 1], we need to output r+2-**e** bits.

It should be **y** bits instead of **e** bits.
> For any x_shrunk in [2^y, 2*2^y - 1], we need to output r+1-**y** bits.
> For any x_shrunk in [2^{y-1}, 2*2^{y-1} - 1], we need to output r+2-**y** bits.

**Reply by kedartatwawadi** [reply]
Thanks again @erickaoshoulin, the typo is now fixed
